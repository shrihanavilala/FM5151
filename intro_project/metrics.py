from intermediate.solutions import utils

def main():
    """Your implementation"""
    metrics_file = open("intro_project/metrics.csv", "w")
    returns_file = open("intro_project/returns.csv", "r")
    with metrics_file, returns_file:
        returns = returns_file.readlines()
        monthly_returns = [yearly_outlook.split(",") for yearly_outlook in returns] #split and ignore header title
        
        metrics_file.write("ticker,avg_return,volatility,alpha,beta\n")
        gspc_returns = [float(line[1]) for line in monthly_returns[1:]] #baseline for market returns
        returns_by_stock = [list(row) for row in zip(*monthly_returns[1:])]
        for stock_title, stock in enumerate(returns_by_stock[1:]):
            numbered = [float(return_) for return_ in stock]
            avg_return = sum(numbered)/len(numbered)
            volatility = utils.stddev(numbered)
            beta = utils.covariance(gspc_returns, numbered)/utils.variance(gspc_returns)
            alpha = avg_return - beta*(sum(gspc_returns)/len(gspc_returns))
            metrics_file.write(f"{monthly_returns[0][stock_title]},{avg_return:0.4f},{volatility:0.4f},{alpha:0.4f},{beta:0.4f}\n")
            
        
        
        
            
        
        
        


if __name__ == "__main__":
    main()
