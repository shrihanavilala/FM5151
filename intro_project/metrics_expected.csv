ticker,avg_return,volatility,alpha,beta
^GSPC,0.0117,0.0521,0.0000,1.0000
NVDA,0.0670,0.1440,0.0474,1.6800
AAPL,0.0282,0.0850,0.0137,1.2417
KO,0.0080,0.0542,0.0010,0.6065
MCD,0.0073,0.0565,-0.0009,0.7046
MS,0.0212,0.0943,0.0050,1.3796
